clang/test/CIR/CodeGen/builtins.cpp

```diff
@@ -56,3 +56,22 @@ int *test_std_addressof2() {
   // LLVM: [[RES:%.*]] = load ptr, ptr [[ADDR]], align 8
   // LLVM: ret ptr [[RES]]
 }
+
+extern "C" char* test_memchr(const char arg[32]) {
+  return __builtin_char_memchr(arg, 123, 32);
```

Comment:
This case is a bit different from others, turns out we already have `MemChrOp`, did you try using that?